### PR TITLE
store connection string in separate class

### PR DESCRIPTION
**Pull request by sophiaso**
## Description
For easier access and more straight-forward abstraction.

### CODE REVIEW

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+public class ConnectionString {
+    private static final String CONN_STRING_REGEXP = "Endpoint=https://([^;]+);Id=([^;]+);Secret=([^;]+)";
+    public static final String ENDPOINT_ERR_MSG = String.format("Connection string does not follow format %s.",
+            CONN_STRING_REGEXP);
+    private static final Pattern CONN_STRING_PATTERN = Pattern.compile(CONN_STRING_REGEXP);
+    public static final String NON_EMPTY_MSG = "%s property should not be null or empty in the connection string of " +
+            "Azure Config Service.";
+
+    private String endpoint;
```
Comment (warrenzhu25) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java]: If no change, better be `final`

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    private String id;
+    private String secret;
+
+    public ConnectionString(String endpoint, String id, String secret) {
```
Comment (warrenzhu25) [inline spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java]: If no direct call of this constructor, better be `private`.

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java (hunk 1 above)
Comment (sophiaso) [reply spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java]: updated.

File: spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java (hunk 2 above)
Comment (sophiaso) [reply spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/resource/ConnectionString.java]: Make sense, for now there is no external access yet, use private first. Updated.

**Pull request merged**